### PR TITLE
Update packaging pipeline torch and cuda version.

### PR DESCRIPTION
**Pull request by AdamLouly**
This PR was made to upgrade the cuda and torch version for packaging pipeline for both nightly and stable pipelines.
Version changes made: 
Stable: 
torch 1.9.0 -> 1.12.1
cuda 10.2 -> 11.6
Nightly:
torch 1.9.0 -> 1.12.1
cuda 11.1 -> 11.6
[Stable Run](https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=230070&view=logs&j=67981f12-2d9a-56a4-530f-a18d19e86536)
[Nightly Run](https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=229980&view=logs&j=67981f12-2d9a-56a4-530f-a18d19e86536)

Deprecated unused dockerfiles for older versions.

### CODE REVIEW

File: torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04 [hunk 1]
```diff
@@ -3,39 +3,43 @@
...
 RUN apt-get install -y software-properties-common wget apt-utils patchelf git libprotobuf-dev protobuf-compiler cmake
 RUN unattended-upgrade
 RUN apt-get autoremove -y
 
+# install miniconda (comes with python 3.9 default)
+RUN apt-get install curl
+ARG CONDA_VERSION=py39_4.10.3
+ARG CONDA_URL=https://repo.anaconda.com/miniconda/Miniconda3-${CONDA_VERSION}-Linux-x86_64.sh
+RUN curl -fSsL --insecure ${CONDA_URL} -o install-conda.sh &&\
+    /bin/bash ./install-conda.sh -b -p /opt/conda &&\
```
Comment (baijumeswani) [inline torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04:23]: Can we instead do a local installation (not in /opt/conda)? You can refer to how it is done in [orrttraining](https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/linux/docker/Dockerfile.ubuntu_gpu_training#L37-L46).

File: torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04 [hunk 2]
```diff
@@ -3,39 +3,43 @@
...
 RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 1
 RUN apt-get install -y python3-pip
 RUN update-alternatives --install /usr/bin/pip pip /usr/bin/pip3 1
 RUN pip install --upgrade pip
 
 # PyTorch
-# TODO: upgrade to latest onnx
-RUN pip install onnx==1.11.0 ninja
-RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+RUN pip install onnx==1.9.0 ninja
```
Comment (baijumeswani) [inline torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04:34]: let's install latest onnx.

File: torch_ort/requirements.txt [hunk 3]
```diff
@@ -4,4 +4,4 @@
 # If a package of a specific version is needed, the package can be installed
 # with version being specified (see Readme.md).
 onnxruntime-training>=1.7.0
-torch>=1.8
+torch>=1.8
```
Comment (baijumeswani) [inline torch_ort/requirements.txt:7]: what changed on this line?

File: torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04 [hunk 4]
```diff
@@ -3,39 +3,43 @@
...
 # PyTorch
-# TODO: upgrade to latest onnx
-RUN pip install onnx==1.11.0 ninja
-RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+RUN pip install onnx ninja
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION} -f https://download.pytorch.org/whl/torch/
 
 # ORT Module
-RUN pip install --pre onnxruntime-training -f https://download.onnxruntime.ai/onnxruntime_nightly_cu102.html
+RUN pip install --pre onnxruntime-training -f https://download.onnxruntime.ai/onnxruntime_nightly_cu116.html
```
Comment (baijumeswani) [inline torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04:38]: Let's change this to the stable release of `onnxruntime-training` 1.13.0. That way the pipeline is ready for the next release.

**Comment by AdamLouly** [comment]
Pipeline Run after the latest changes: [Here](https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=230908&view=results)

File: torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04 [hunk 5]
```diff
@@ -3,39 +3,45 @@
...
 # PyTorch
-# TODO: upgrade to latest onnx
-RUN pip install onnx==1.11.0 ninja
-RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION}+${TORCH_CUDA_VERSION} -f https://download.pytorch.org/whl/torch_stable.html
+RUN pip install onnx ninja
+RUN pip install torch==${TORCH_VERSION}+${TORCH_CUDA_VERSION} torchvision==${TORCHVISION_VERSION} -f https://download.pytorch.org/whl/torch/
 
 # ORT Module
-RUN pip install --pre onnxruntime-training -f https://download.onnxruntime.ai/onnxruntime_nightly_cu102.html
+RUN pip install --pre onnxruntime-training -f https://download.onnxruntime.ai/onnxruntime_stable_cu116.html
```
Comment (baijumeswani) [inline torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04:40]: this should not be `--pre`.

**Review by baijumeswani — APPROVED** [review]

File: torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04 [hunk 6]
```diff
@@ -0,0 +1,47 @@
+# ONNX Runtime Training Module for PyTorch
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# CUDA development image for building sources
+FROM nvidia/cuda:11.6.2-cudnn8-devel-ubuntu18.04 as builder
+ARG TORCH_CUDA_VERSION=cu116
+ARG ONNXRUNTIME_TRAINING_VERSION=1.12.0
```
Comment (askhade) [inline torch_ort/docker/Dockerfile.ort-default-stable-torch1.12.1-onnxruntime190-cu116-cudnn8-devel-ubuntu18.04:8]: unused variable?

**Review by askhade — APPROVED** [review]

**Comment by AdamLouly** [comment]
Thanks for the review

**Pull request merged**